fuzz/staking/src/main.rs

```diff
@@ -0,0 +1,435 @@
+    RuntimeType,
+};
+use sp_runtime::{BuildStorage, Percent};
+use sp_state_machine::BasicExternalities;
+use std::collections::BTreeMap;
+use subspace_runtime_primitives::AI3;
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct FuzzData {
+    /// 5 epochs with N epochs skipped
+    pub epochs: [(u8, Epoch); 5],
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct Epoch {
+    /// 5 actions split between 5 users
+    actions: [(u8, FuzzAction); 5],
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+enum FuzzAction {
+    RegisterOperator {
+        amount: u16,
+    },
+    NominateOperator {
+        operator_id: u8,
+        amount: u16,
+    },
+    DeregisterOperator {
+        operator_id: u64,
+    },
+    WithdrawStake {
+        nominator_id: u8,
+        operator_id: u8,
+        shares: u16,
+    },
+    UnlockFunds {
+        operator_id: u8,
+        nominator_id: u8,
+    },
+    UnlockNominator {
+        operator_id: u8,
+        nominator_id: u8,
+    },
+    MarkOperatorsAsSlashed {
+        operator_id: u8,
+        slash_reason: u8, // 0 for InvalidBundle, 1 for BadExecutionReceipt
+    },
+    MarkInvalidBundleAuthors {
+        operator_id: u8,
+    },
+    UnmarkInvalidBundleAuthors {
+        operator_id: u8,
+        er_id: u8,
+    },
+    RewardOperator {
+        operator_id: u8,
+        amount: u16,
+    },
+    SlashOperator,
+}
+
+fn create_genesis_storage(accounts: &[AccountId], mint: u128) -> Storage {
+    let raw_genesis_storage = RawGenesis::dummy(vec![1, 2, 3, 4]).encode();
+    let pair = OperatorPair::from_seed(&[*accounts.first().unwrap() as u8; 32]);
+    pallet_domains::tests::RuntimeGenesisConfig {
+        balances: BalancesConfig {
+            balances: accounts.iter().cloned().map(|k| (k, mint)).collect(),
+        },
+        domains: pallet_domains::tests::DomainsConfig {
+            genesis_domains: vec![GenesisDomain {
+                runtime_name: "evm".to_owned(),
+                runtime_type: RuntimeType::Evm,
+                runtime_version: Default::default(),
+                raw_genesis_storage,
+                owner_account_id: *accounts.first().unwrap(),
+                domain_name: "evm-domain".to_owned(),
+                bundle_slot_probability: (1, 1),
+                operator_allow_list: OperatorAllowList::Anyone,
+                signing_key: pair.public(),
+                minimum_nominator_stake: 100 * AI3,
+                nomination_tax: Percent::from_percent(5),
+                initial_balances: vec![],
+                domain_runtime_info: (DEFAULT_EVM_CHAIN_ID, Default::default()).into(),
+            }],
+            permissioned_action_allowed_by: Some(PermissionedActionAllowedBy::Anyone),
+        },
+        subspace: Default::default(),
+        system: Default::default(),
+    }
+    .build_storage()
+    .unwrap()
+}
+
+fn main() {
```

Comment:
This fuzzing harness isn't tested in CI, so any other PR could accidentally break it. We've had similar issues with the runtime benchmarks in the past.

Task for Teor:
- before this PR merges, add a single fuzzer run in CI, to make sure it works
- add it to the rust-all job

Here's how that works for benchmarks:
https://github.com/autonomys/subspace/blob/3d0d436c9c95eb51a56af199b8e1c461f7602ab1/.github/workflows/rust.yml#L329-L331